### ISSUE TITLE
Consolidated Cases - Trial Session Information Page

### ISSUE THREAD

**Issue by cholly75** [ustaxcourt/ef-cms#8250]
As a court user, so that I can easily tell which cases are grouped together for consolidation in a trial session, I need the Trial Session case listing to display cases in the same consolidated group together.

Currently on the Trial Session page the cases that have been calendared for the session are listed in ascending docket number order on all views (Open, Inactive, and All). When consolidated cases are a part of the session, the cases can fall anywhere on the list according to that sort order. This leads to confusion and manual work required to find all the cases in a particular group.

We would like to implement the consolidated iconography and grouping on the Trial Session page when listing the cases in these views.

## Pre-Conditions
depends on ustaxcourt/ef-cms#6971

## Acceptance Criteria
* On the Open, Inactive, or All view on the Session Information page, cases belonging to a consolidated group are indicated by an icon indicating whether or not the case is the Lead case or a Member case.
* On the Open, Inactive, or All view on the Session Information page, cases belonging to the same consolidated group are grouped together in the case listing, with the Lead case listed first followed by the member cases slightly indented underneath in ascending docket number order (i.e., oldest -> newest)
  - Example: Cases 123-20, 234-20, and 345-21 are in a consolidated group together, with 123-20 being the Lead case. The
     case listing for these cases when calendared would be ordered as such:
       123-15
       123-16
       123-20
         -234-20
         -345-21
       150-20
       123-22
* Case listings should retain default sorting order
* Consolidated Group overall placement on the list is determined by the Lead Case in the group
* If member cases of the group are on the view but the Lead Case is NOT, do not indent the member cases in the list.

## Story Definition of Ready (Created on 9/26/21)
The following criteria must be met in order for the user story to be picked up by the Flexion development team.
The user story must: 
- [ ]  Be immediately actionable for the development team
- [ ]  Be focused on users
- [ ]  Include a narrative. “As a ____, so that I can ______, I need _____.”
- [ ]  Include user-centric acceptance criteria that expresses the users' needs
- [ ]  Include user-centric testing scenarios
- [ ]  Include preconditions (story dependencies)
- [ ]  Not include design or technical solutions
- [ ]  Include non-standard testing, deployment, documentation, browser, mobile or security requirements, as well as any requirements for 3rd-party integrations

Process: Flexion developers and designers will test if the story meets acceptance criteria and test cases in Flexion dev and staging environments (“standard testing”). If additional acceptance criteria or testing scenarios are discovered while the story is in progress, a new story should be created, added to the backlog and prioritized by the product owner. 

## Definition of Done (Updated 5-19-22)
**Product Owner**
 - [ ]  Acceptance criteria have been met and validated on the Court's migration environment
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**UX**
 - [ ] Business test scenarios have been refined to meet all acceptance criteria 
 - [ ] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)

**Engineering**
 - [x] Automated test scripts have been written, including visual tests for newly added PDFs
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] Verify that language for docket record for internal users and external users is identical
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Interactors should validate entities before calling persistence methods
 - [x] Code refactored for clarity and to remove any known technical debt
 - [x] Acceptance criteria for the story has been met
 - [x] If there are special instructions in order to deploy into the next environment, add them as a comment in the story
 - [ ] Reviewed by UX on a deployed environment.
 - [ ] Reviewed by PO on a deployed environment. Can be deployed to the Court's `test` environment if prod-like data is required. Otherwise deployed to any `experimental` environment.
 - [ ] Deployed to the Court's `staging` environment.

**Comment by swongCO** [comment]
Refer to https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/9535 for implementation

**Comment by swongCO** [comment]
Refer to https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/9535 for implementation

**Comment by ttlenard** [comment]
## Test Cases


#### 1) On the Open Cases tab on the Trial Session information page, a Consolidated Group of cases has iconography indicating if a case is the lead or a member case; Consolidated group is sorted appropriately. 

Preconditions: You will need to ensure that there are a couple different consolidated groups of cases that have been calendared for a trial session. 

- Log in as a Court user
- Click on the Trial Sessions tab
- From the Open tab, Click on the city/state of the trial session that has the consolidated group of cases calendared.
- Scroll down to the list of cases on the Session information page.
- Be sure that you are on the Open Cases tab.


Expected Results:
- [x] The lead case in the consolidated group has the Lead case indicator icon
- [x] The member cases in the consolidated group have the member case indicator icon
- [x] The member cases are indented slightly
- [x] The Cases in the consolidated group are grouped in ascending order (oldest to newest)
- [x] The default sorting order of the case lists are not changed (sorted by ascending docket numbers)
- [x] The overall placement of the consolidated group on the Open Cases list is determined by the Lead Case in the Group.
- [x] Be sure to also check the formatting for the placement of the iconography/column data when there is also the "manually added" icon (The green battery). Columns should remain in alignment when this symbol is present.



#### 2) On the Inactive Cases tab on the Trial Session information page, a Consolidated Group of cases has iconography indicating if a case is the lead or a member case; Consolidated group is sorted appropriately. 

Preconditions: You will need change the status of one of the consolidated groups of cases that have been calendared for a trial session (change the status on all of the cases). (Docket clerks can do this. Continue the cases generally, enter a decision, etc.)

- Log in as a Court user
- Click on the Trial Sessions tab
- From the Open tab, Click on the city/state of the trial session that has the consolidated group of cases calendared.
- Scroll down to the list of cases on the Session information page.
- Be sure that you are on the Inactive Cases tab.


Expected Results:
- [x] The lead case in the consolidated group has the Lead case indicator icon
- [x] The member cases in the consolidated group have the member case indicator icon
- [x] The member cases are indented slightly
- [x] The Cases in the consolidated group are grouped in ascending order (oldest to newest)
- [x] The default sorting order of the case lists are not changed (sorted by ascending docket numbers)
- [x] The overall placement of the consolidated group on the Inactive Cases list is determined by the Lead Case in the Group.
~~- [ ] Be sure to also check the formatting for the placement of the iconography/column data when there is also the "manually added" icon (The green battery). Columns should remain in alignment when this symbol is present.~~



#### 3) On the All Cases tab on the Trial Session information page, a Consolidated Group of cases has iconography indicating if a case is the lead or a member case; Consolidated group is sorted appropriately. 

- Continuing from test case 1 and 2
- Click on the All cases tab

Expected Results:
- [x] The lead case in the consolidated group has the Lead case indicator icon
- [x] The member cases in the consolidated group have the member case indicator icon
- [x] The member cases are indented slightly
- [x] The Cases in the consolidated group are grouped in ascending order (oldest to newest)
- [x] The default sorting order of the case lists are not changed (sorted by ascending docket numbers)
- [x] The overall placement of the consolidated group on the All Cases list is determined by the Lead Case in the Group.
- [x] Be sure to also check the formatting for the placement of the iconography/column data when there is also the "manually added" icon (The green battery). Columns should remain in alignment when this symbol is present.


#### 4) Docket Clerk updates the case status of the Lead case in a consolidated group; Consolidated iconography is retained; Open Cases, Inactive Cases, and All Cases list views are updated appropriately.

Preconditions: You will need to ensure that there are a couple different consolidated groups of cases that have been calendared for a trial session. 

- Log in as a docket clerk
- Navigate to the lead case in the consolidated group of cases that is still on the Open case list.
- Click on Case Information tab
- Click on the Edit button in the Trial Card, and remove the case from the trial
- Navigate to the trial session that this case was calendared on.

Expected Results
- [x] On the Open Case List
     - [x] The lead case that you just changed the status for will no longer be on the Open Case List.
     - [x] The member cases in the consolidated group have the member case indicator icon
     - [x] The member cases that are with the group are no longer indented on the open case list, since the lead case is no longer Open.
     - [x] The member cases that are with the group are no longer grouped together, and are now sorted in the list by the default sorting order (ascending docket numbers) 
- [x] On the Inactive Case List
	- [x] The lead case in the group is now on the Inactive Cases List.
	- [x] The lead case in the consolidated group has the Lead case indicator icon.
	- [x] The lead case in the group does not have any member cases beneath it on the Inactive Cases Tab.
- [x] On the All Cases list:
     - [x] The lead case in the consolidated group has the Lead case indicator icon
     - [x] The member cases in the consolidated group have the member case indicator icon
     - [x] The member cases are indented slightly
     - [x] The Cases in the consolidated group are grouped in ascending order (oldest to newest)
     - [x] The default sorting order of the case lists are not changed (sorted by ascending docket numbers)
     - [x] The overall placement of the consolidated group on the All Cases list is determined by the Lead Case in the Group.
     - [x] Be sure to also check the formatting for the placement of the iconography/column data when there is also the "manually added" icon (The green battery). Columns should remain in alignment when this symbol is present.
     - [x] The Case Status for the lead case in the group reflects the status you changed it to.  The member cases Case status should remain "Calendared".



#### 5) Docket Clerk updates the case status of a member case in a consolidated group; Consolidated iconography is retained; Open Cases, Inactive Cases, and All Cases list views are updated appropriately.

This test has some set up:
- Continuing from the previous test case, as a docket clerk, manually add the lead case back on the trial session by navigating to the case, clicking on Case information tab, then clicking add to trial.
- Do a quick check to ensure that the lead case is back on the Open case list, and the consolidated case iconography and sorting is reflected appropriately (as it was in test case 1).
- Next, navigate to a member case in the consolidated group of cases that is on the Open case list.
- Click on Case Information tab
- Click on the Edit button in the Trial Card, and remove the case from the trial.
- Navigate to the trial session that this case was calendared on.

Expected Results:
- [x] On the Open Cases List:
     - [x] The member case that you just changed the status for will no longer be on the Open Case List.
     - [x] The lead case in the consolidated group still has the Lead case indicator icon.
     - [x] The other member cases in the consolidated group still appear indented slightly under the lead case.
     - [x] The Cases in the consolidated group are grouped in ascending order (oldest to newest)
     - [x] The default sorting order of the case lists are not changed (sorted by ascending docket numbers)
- [x] On the Inactive Case List:
	- [x] The member case is now on the Inactive Case list
	- [x] The member case has the member case indicator icon
	- [x] The member case is listed on the inactive case list in the default sorting order (ascending docket number)
	- [x] The member case on the inactive case list is not indented.
- [x] On the All Cases list:
     - [x] The lead case in the consolidated group has the Lead case indicator icon
     - [x] The member cases in the consolidated group have the member case indicator icon
     - [x] The member cases are indented slightly (including the member case that you changed the status for)
     - [x] The Cases in the consolidated group are grouped in ascending order (oldest to newest)
     - [x] The default sorting order of the case lists are not changed (sorted by ascending docket numbers)
     - [x] The overall placement of the consolidated group on the All Cases list is determined by the Lead Case in the Group.
     - [x] Be sure to also check the formatting for the placement of the iconography/column data when there is also the "manually added" icon (The green battery). Columns should remain in alignment when this symbol is present.
     - [x] The Case Status for the member case in the group reflects the status you changed it to.  The Lead and all other member cases Case status should remain "Calendared".



#### 6) Docket Clerk unconsolidates a member case in the consolidated group; Consolidated iconography and sorting on the case lists are updated appropriately.

- Log in as a Docket Clerk
- Navigate to one of the member cases from the consolidated groups that you have been using in the previous test cases (case is calendared for trial).
- Click on the case information tab
- Unconsolidate the member case from the group by clicking the Remove cases link on the Consolidated Cases Card.
- Click on the City, State of the trial session from the case information tab.
- When the Trial Session Information page opens, scroll down to the Open Cases tab.

Expected Results:
- [x] The member case that you just unconsolidated from the group is sorted in the list by docket number, and is no longer grouped with the consolidated group.
- [x] The member case icon is no longer present

* Expected results should be also be reviewed on the All Cases List.

#### 7) Docket Clerk unconsolidates the lead case in the consolidated group; Consolidated iconography and sorting on the case lists are updated appropriately.

- Log in as a Docket Clerk
- Navigate to one of the Lead cases from the consolidated groups that you have been using in the previous test cases (case is calendared for trial).
- Click on the case information tab
- Take note of the next case in ascending order that is currently a member case (this case is going to change to the new lead case after this test)
- Unconsolidate the Lead case from the group by clicking the Remove cases link on the Consolidated Cases Card.
- Click on the City, State of the trial session from the case information tab.
- When the Trial Session Information page opens, scroll down to the Open Cases tab.

Expected Results:
- [x] The Lead case that you just unconsolidated from the group is sorted in the list by docket number, and is no longer grouped with the consolidated group.
- [x] The Lead case icon is no longer present for this docket number
- [x] The lead case icon is now present on the member case that you noted earlier in this test.

* Expected results should be also be reviewed on the All Cases List.



#### 8) Docket Clerk adds a new case from the Open cases list to the consolidated group as a member case; the new member case displays consolidated status appropriately on the Open Case List

- Before doing this test, be sure to take note of a case from the Open case list that has a newer docket number than the lead case in the Consolidated group.
- Log in as a docket clerk
- Navigate to the case that you took note of
- Click on the Case information tab
- Click on the Add Cases Link, and add the case to the consolidated group
- Click on the city/state link of the trial session from the case information tab.
- When the Trial Session Information page opens, scroll down to the Open Cases tab.

Expected Results:
- [x] The newly added case displays with the consolidated group, in ascending order
- [x] The indicator icon should be a member case icon
- [x] The member case is indented slightly



#### 9) Docket Clerk adds a new case from the Open case list to the consolidated group as the new lead case; the new lead case displays consolidated status appropriately on the Open Case List

- Before doing this test, be sure to take note of a case from the Open case list that has an older docket number than the current lead case in the Consolidated group.
- Log in as a docket clerk
- Navigate to the case that you took note of
- Click on the Case information tab
- Click on the Add Cases Link, and add the case to the consolidated group
- Click on the city/state link of the trial session from the case information tab.
- When the Trial Session Information page opens, scroll down to the Open Cases tab.

Expected Results:
- [x] The newly added case displays with the consolidated group, in ascending order
- [x] The indicator icon should be the Lead case icon
- [x] The previous Lead case is now a member case, has the member case icon, and is indented slightly

**Comment by jhansen-flexion** [comment]
Blocked because it relies on depends on [#3572](https://app.zenhub.com/workspaces/flexionef-cms-5bbe4bed4b5806bc2bec65d3/issues/flexion/ef-cms/3572)
